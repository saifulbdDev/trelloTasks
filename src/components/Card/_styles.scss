@import "../../styles/sass/variables.scss";
@import "../../styles/sass/mixins.scss";

.card {
  position: relative;
  font-size: 1.4rem;
  font-weight: 400;
  line-height: 1.3;
  background-color: $white;
  padding: 0.65rem 0.6rem;
  color: #4d4d4d;
  border-bottom: 0.1rem solid #ccc;
  border-radius: 0.3rem;
  margin-bottom: 0.6rem;
  word-wrap: break-word;
  cursor: pointer;
  overflow: hidden;
  margin: 0;
  &:last-of-type {
    margin-bottom: 0;
  }
  &:hover {
    background-color: #eee;
    .card-actions {
      opacity: 0.7;
    }
  }
  .card-actions {
    display: flex;
    align-items: center;
    position: absolute;
    top: 0;
    bottom: 0;
    right: 0;
    left: auto;
    opacity: 0;
    button{
      margin-left: 0.5rem;
    }
  }

  .card-body {
    text-align: left;
    overflow: hidden;
    text-overflow: ellipsis;
    word-wrap: break-word;
  }
}

.isDragged,
.isDragged:hover {
  cursor: move;
  border: 1px dashed grey;
}

.card-header {
  outline: 1px solid $blue;
}
