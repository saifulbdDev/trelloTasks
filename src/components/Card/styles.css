.card {
  font-size: 1.4rem;
  font-weight: 400;
  line-height: 1.3;
  background-color: #fff;
  padding: 0.65rem 0.6rem;
  color: #4d4d4d;
  border-bottom: 0.1rem solid #ccc;
  border-radius: 0.3rem;
  margin-bottom: 0.6rem;
  word-wrap: break-word;
  cursor: pointer;
  margin: 0;
}

.card:last-of-type {
  margin-bottom: 0;
}

.card:hover {
  background-color: #eee;
}

.card .card-body {
  text-align: left;
  overflow: hidden;
  text-overflow: ellipsis;
}

.isDragged, .isDragged:hover {
  cursor: move;
  border: 1px dashed grey;
}

.card-header {
  outline: 1px solid #1010ff;
}

.button-remove {
  background: transparent;
  font-size: 1.2em;
  float: right;
  outline: none;
  color: white;
  border: none;
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0;
  opacity: 0;
}

.button-remove:hover {
  opacity: 0.7;
}

textarea {
  overflow: hidden;
  width: 80%;
}
/*# sourceMappingURL=styles.css.map */