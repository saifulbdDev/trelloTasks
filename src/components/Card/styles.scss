@import "../../styles/sass/variables.scss";
@import "../../styles/sass/mixins.scss";

.card {
  max-width: 100%;
  margin: 5 * $offsetWidth auto;
  color: $black;
  text-align: center;
  position: relative;
  background-color: $white;
  @include box-shadow(
    0 $offsetWidth 3 * $offsetWidth -$offsetWidth rgba($lightgrey, 0.1)
  );
}

.Card:hover {
  cursor: pointer;
}

.isDragged, .isDragged:hover {
  cursor: move;
  border: 1px dashed grey;
}

.card-header {
  outline: 1px solid $blue;
}

.card-body {
  padding: 2 * $offsetWidth;
  text-align: center;
  overflow: hidden;
  text-overflow: ellipsis;
}


.button-remove {
  @include button();
  position: absolute;
  top: 0;
  right: 0;
  opacity: 0;
  &:hover {
    opacity: 0.7;
  }
}

textarea {
  overflow: hidden;
  width: 80%;
}
