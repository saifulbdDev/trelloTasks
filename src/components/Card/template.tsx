import React from "react";
import "./styles.scss";

interface ICardProps {
  category: string;
  categoryTasks?: any;
  createCard?: any;
  removeCard?: any;
  updateCard?: any;
  editable?: boolean;
  text?: string;
  value?: string;
  id: number;
}

interface ICardState {
  id?: number;
  text?: string;
  value?: string;
  editable?: boolean;
}

export const Card: React.FC<ICardProps> = (props) => {
  const [state, setState] = React.useState<ICardState>(props);
  const [isDragged, setIsDragged] = React.useState<boolean>(false);

  React.useEffect(() => {
    setState({ editable: false, value: "", text: "", id: 0 });
  }, [props.category]);

  const handleOnEdit = () => {
    setState({ editable: true });
  };

  const handleOnRemove = () => {
    props.removeCard(props.id);
  };

  const handleOnBlur = (event: { target: { value: string; }; }) => {
    const cleanedValue = event.target.value.trim();
    if (cleanedValue !== "") {
      setState({ editable: false, value: event.target.value });
      props.updateCard({ id: props.id, text: state.text });
    }
  };

  const handleChange = (event: any) => {
    setState({ value: event.target.value });
  };

  const handleOnDragStart = (ev: any, id: number) => {
    ev.dataTransfer.setData("id", id);
    ev.dataTransfer.setData("sourceCategory", props.category);
    setIsDragged(true);
  };

  const handleOnDragStop = () => {
    setIsDragged(false);
  }

  const getInputField = () => {
    return (
      <textarea
        value={state.value}
        onChange={handleChange}
        onBlur={handleOnBlur}
      />
    );
  }

  const content = state.editable
    ? getInputField()
    : props.text;

  return (
    <div
      className={`card ${props.category}` + (isDragged ? "isDragged" : "")}
      onDragStart={(e) => handleOnDragStart(e, props.id)}
      onDragEnd={handleOnDragStop}
      draggable
    >
      <button className="button-remove" onClick={handleOnRemove}>âœ–</button>
      <div className="card-body" onClick={handleOnEdit}>
        {content}
      </div>
    </div>
  );

}