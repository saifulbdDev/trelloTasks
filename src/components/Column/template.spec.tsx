import React from "react";
import { shallow } from "enzyme";
import { Column } from "./";
import { Card } from "../Card";

describe("Column", () => {
  const props = {
    categoryTasks: [],
    category: "abc",
    createCard: jest.fn(),
    moveCard: jest.fn(),
    updateCard: jest.fn(),
    removeCardFromColumn: jest.fn(),
  };

  it("renders properly", () => {
    const wrapper = shallow(<Column {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render different cards ", () => {
    const mockCategoryTasks = ["a", "b", "c"];
    const wrapper = shallow(<Column {...props} categoryTasks={mockCategoryTasks} />);
    expect(wrapper.find(Card)).toHaveLength(3);
  });
});
