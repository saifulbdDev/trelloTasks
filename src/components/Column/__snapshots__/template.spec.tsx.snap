// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column renders properly 1`] = `ShallowWrapper {}`;
